torchtext/datasets/amazonreviewpolarity.py

```diff
@@ -40,11 +40,25 @@ def AmazonReviewPolarity(root: str, split: Union[Tuple[str], str]):
         raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
 
     url_dp = IterableWrapper([URL])
-    cache_dp = url_dp.on_disk_cache(
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-    cache_dp = FileOpener(cache_dp, mode="b")
-    extracted_files = cache_dp.read_from_tar()
-    filter_extracted_files = extracted_files.filter(lambda x: _EXTRACTED_FILES[split] in x[0])
-    return filter_extracted_files.parse_csv().map(fn=lambda t: (int(t[0]), ' '.join(t[1:])))
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    def extracted_filepath_fn(x):
```

Comment:
Should we move this to `torchtext/data/datasets_utils.py` since we would likely want to reuse this across all of our datasets?

Reply:
I am not so sure, we might need to use different functions for different datasets depending on how to compression is set.

Reply:
Gotcha let's figure out where to put this function as we start updating the caching across our other datasets.

Reply:
I can confirm I've used this identical function in a few of my own double-caching migrations so it might be worth pulling out.

Reply:
Actually, While working in parallel here https://github.com/pytorch/data/pull/169 I realize we do not need to explicitly create the directory. I guess, we can get rid of this function all together. Refer to this comment thread (https://github.com/pytorch/data/pull/169#discussion_r787228019)

Reply:
BTW, the main idea behind this function is to look for whether the file already exist or not (refer here https://github.com/pytorch/data/pull/169#discussion_r787229179). So I guess, which files to check would depend and vary for different datasets.